btrace-agent/src/main/java/org/openjdk/btrace/agent/Main.java

```diff
@@ -750,6 +750,10 @@ public FileVisitResult postVisitDirectory(Path dir, IOException exc)
   }
 
   private static boolean loadBTraceScript(String filePath, boolean traceToStdOut) {
+    if (!filePath.endsWith(".class")) {
```

Comment:
Could you modify this to look at the magic bytes instead?
For Java class files it would be `0xCAFEBABE` and for the BTrace probe-pack it is `0xBACECACA`. 
The problem with only extension check is that the BTrace probe-pack can have any extension name - it is not really standardized. For simplicity it defaults to `.class` but it is possible to specify some other extension so you eg. don't confuse the classloader if the BTrace probe pack files are put together with regular class files.

Reply:
How would I check for the magic bytes? Is it as simple as opening the file, read the first few bytes and compare them to the mentioned 2 magic bytes?

Reply:
Oh, I am sorry, I completely missed the notification about the activity here :(
Yes, it is exactly the way you are describing. It is 4 first bytes of the file which is containing the magic number.